distribution/metadata/v2_metadata_service.go

```diff
@@ -151,6 +158,9 @@ func (serv *v2MetadataService) GetDiffID(dgst digest.Digest) (layer.DiffID, erro
 // Add associates metadata with a layer DiffID. If too many metadata entries are
 // present, the oldest one is dropped.
 func (serv *v2MetadataService) Add(diffID layer.DiffID, metadata V2Metadata) error {
+	if serv.store == nil {
+		return nil
```

Comment:
I think I understand why the getters return an error but the setters silently do nothing, but it might be good to have a comment explaining it.